pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -120,3 +124,41 @@ func indexedDomainInterfaces(domain *api.Domain) map[string]api.Interface {
 	}
 	return domainInterfaces
 }
+
+// withNetworkIfacesResources adds network interfaces as placeholders to the domain spec
+// to trigger the addition of the dependent resources/devices (e.g. PCIe).
+// As its last step, it reads the generated configuration and removed the network interfaces
+// so none will be created with the domain creation.
+// The dependent devices are left in the configuration, to allow future hotplug.
+func withNetworkIfacesResources(vmi *v1.VirtualMachineInstance, domainSpec *api.DomainSpec, f func(v *v1.VirtualMachineInstance, s *api.DomainSpec) (cli.VirDomain, error)) (cli.VirDomain, error) {
+	domainSpecWithIfacesResource := domainSpec.DeepCopy()
+	requests := resource.ExtendedResourceList{ResourceList: vmi.Spec.Domain.Resources.Requests}
+	reqInterfaces := int(requests.Interface().Value())
+	for i := 0; i < (reqInterfaces - len(vmi.Spec.Domain.Devices.Interfaces)); i++ {
+		domainSpecWithIfacesResource.Devices.Interfaces = append(domainSpecWithIfacesResource.Devices.Interfaces, api.Interface{
+			Type:  "ethernet",
+			Model: &api.Model{Type: "virtio-non-transitional"},
+			Target: &api.InterfaceTarget{
+				Device:  fmt.Sprintf("placeholder-%d", i),
+				Managed: "no",
+			},
+		})
+	}
+	dom, err := f(vmi, domainSpecWithIfacesResource)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(domainSpec.Devices.Interfaces) == len(domainSpecWithIfacesResource.Devices.Interfaces) {
+		return dom, nil
+	}
+
+	domainSpecWithoutIfacePlaceholders, err := util.GetDomainSpecWithFlags(dom, libvirt.DOMAIN_XML_INACTIVE)
+	if err != nil {
+		return nil, err
+	}
+	domainSpecWithoutIfacePlaceholders.Devices.Interfaces = domainSpec.Devices.Interfaces
+	domainSpecWithoutIfacePlaceholders.DeepCopyInto(domainSpec)
+
+	return f(vmi, domainSpecWithoutIfacePlaceholders)
```

Comment:
can you export this part of the code to a well named function ? 

I think it would help me allow the reader to better understand the goal of this section of the code.